src/persist-client/src/impl/machine.rs

```diff
@@ -758,15 +761,86 @@ mod tests {
 
             let state = Arc::new(Mutex::new(DatadrivenState::default()));
             let cpu_heavy_runtime = Arc::new(CpuHeavyRuntime::new());
+            let write = Arc::new(Mutex::new(
+                client
+                    .open_writer::<String, (), u64, i64>(shard_id)
+                    .await
+                    .expect("invalid shard types"),
+            ));
 
             f.run_async(move |tc| {
                 let shard_id = shard_id.clone();
                 let client = client.clone();
                 let state = Arc::clone(&state);
                 let cpu_heavy_runtime = Arc::clone(&cpu_heavy_runtime);
+                let write = Arc::clone(&write);
                 async move {
                     let mut state = state.lock().await;
+
                     match tc.directive.as_str() {
+                        // Returns whether a blob contains a given batch. The batch
+                        // must contain only a single key.
+                        "blob-batch-exists" => {
+                            let input = get_arg(&tc.args, "input").expect("missing input");
+                            let batch = state.batches.get(input).expect("unknown batch").clone();
+                            let key = batch.keys.get(0).expect("unknown batch part");
+
+                            assert_eq!(batch.keys.len(), 1);
+
+                            let blob = client
+                                .blob
+                                .get(&key.complete(&shard_id))
+                                .await
+                                .expect("blob read failed");
+
+                            format!("{}\n", blob.is_some())
+                        }
+                        // Scans consensus and returns all states with their SeqNos
+                        // and which batches they reference
+                        "consensus-scan" => {
+                            let from =
+                                SeqNo(get_u64(&tc.args, "from_seqno").expect("missing from_seqno"));
+                            let res = client.consensus.scan(&shard_id.to_string(), from).await;
+
+                            let mut s = String::new();
+                            match res {
```

Comment:
I prefer a style (and try to enforce it throughout persist) where error handling happens with early-returns/continues. so here, i'd do something like `let states = match` and have the error branch do the println and an early return. this allows us to dedent the `for persisted_state in states` loop a couple levels and also means as you read through code, you can reason about the error handling as you go instead of jumping up and down